sdk/test-utils/recorder-new/src/recorderAADCredential.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, TokenCredential } from "@azure/core-auth";
+
+/**
+ * `TokenCredential` implementation for playback.
+ * If your regular AAD credentials don't take the recorder httpClient option, the AAD traffic won't be recorded.
+ * In this case, you'll need to bypass the AAD requests with no-op.
+ *
+ * Using this NoOpCredential as your credential in playback mode would help you bypass the AAD traffic.
+ */
+export class NoOpCredential implements TokenCredential {
+  getToken(): Promise<AccessToken> {
+    return Promise.resolve({ token: "TEST TOKEN PLAYBACK", expiresOnTimestamp: 86400 * 1000 });
```

Comment:
This expiration date means the token is expired! is that intended? It could cause undesired behaviors in the pipelines.

Reply:
I just kept this to satisfy the return type

Reply:
Do you want me to multiply it by a million more?

Reply:
I don't know whether "TEST TOKEN PLAYBACK" would trigger credscan error or not, but "SecretPlaceholder" will be ignored for sure.

```suggestion
    return Promise.resolve({ token: "SecretPlaceholder", expiresOnTimestamp: 86400 * 1000 });
```

Reply:
This is a good point. @HarshaNalluru what about `Date.now() + 86400 * 1000)`? I think that's in the spirit of what you're trying to accomplish. The core-rest-pipeline token cycler will probably have a heart attack if we feed it an expired token over and over again.

Reply:
Well, I guess it's using the pipeline now? So now my question is why doesn't the pipeline throw a fit over this?

Reply:
@sadasant ^

Reply:
I’m not sure why it isn’t breaking, but I believe we should use an unexpired date 😆

Based on https://262.ecma-international.org/11.0/#sec-time-values-and-time-range

Seems like the maximum date is `new Date(8640000000000000)` 😁 how does that sound?

Reply:
`Date.now() + 86400 * 1000` should be enough, I'll go with that,